crates/astria-composer/src/executor/mod.rs

```diff
@@ -276,7 +285,93 @@ impl Executor {
+            }
+            Err(reason) => {
+                error!(%reason, "shutting down");
+            }
+        };
+
+        let shutdown_logic = async {
+            info!("setting executor status to disconnected");
+            self.status
+                .send_modify(|status| status.is_connected = false);
+
+            info!("draining sequence actions from the executor receiver channel");
+            // drain the receiver channel
+            while let Ok(seq_action) = self.serialized_rollup_transactions.try_recv() {
+                let rollup_id = seq_action.rollup_id;
+                if let Err(e) = bundle_factory.try_push(seq_action) {
+                    warn!(
+                        rollup_id = %rollup_id,
+                        error = &e as &StdError,
+                        "failed to bundle sequence action: too large. sequence action is dropped."
+                    );
+                }
+            }
+
+            info!("draining remaining bundles from bundle factory");
+            // when shutting down, drain all the remaining bundles and submit to the sequencer
+            // to avoid any bundle loss.
+            let mut bundles_to_drain: Vec<SizedBundle> = vec![];
+            // drain all the bundles from the factory
+            loop {
+                let bundle = bundle_factory.pop_now();
+                if bundle.is_empty() {
+                    break;
+                }
+                bundles_to_drain.push(bundle);
+            }
+            let num_bundles_to_drain = bundles_to_drain.len();
+            if num_bundles_to_drain == 0 {
+                info!("no bundles to drain; executor shutdown tasks completed");
+                return Ok(());
+            }
+
+            info!(
+                "submitting {} bundles to sequencer to drain",
+                num_bundles_to_drain
+            );
+            // we need to submit the bundles sequentially since that is the guarantee
+            // the bundle factory gives to the end users.
+            for bundle in bundles_to_drain {
+                match self.submit_bundle(nonce, bundle).await {
+                    Ok(new_nonce) => {
+                        debug!("drained bundle successfully; new nonce: {}", new_nonce);
+                        nonce = new_nonce;
+                    }
+                    Err(e) => {
+                        error!(%e, "failed submitting bundle to sequencer during shutdown; \
+                                aborting shutdown");
+                        // if we can't drain a bundle after multiple retries, we can abort
+                        // the shutdown process
+                        return Err(e);
+                    }
+                }
+            }
+            info!("drained {} bundles successfully", num_bundles_to_drain);
+
+            Ok(())
+        };
+
+        match tokio::time::timeout(Duration::from_secs(13), shutdown_logic).await {
```

Comment:
I think we should avoid using a timeout here _and_ for the `executor_task_handle` future in the composer's `ShutdownInfo::run`.  Seems like they're both defending against the same thing - i.e. this executor task not returning in time, so we can probably remove the timeout from the composer (maybe with a comment saying it already has protection against running for too long?)  If the composer's timeout happens to execute before we finish up shutting down in the executor, then we lose the benefit of changes here, i.e. we'd potentially not send all the bundles, or report unsent bundles.

If we do want to keep both timeouts, then I'd suggest providing a `pub(super)` const for the timeout duration in the executor module, and use that (+ some wiggle room) in the composer.  But that seems suboptimal to me.

Reply:
I think we can just keep it in executor, it could make more sense. my thinking was that we could have the lower level timeout task(i.e executor) be much bigger and the 13s would be attributed to only that specific shutdown logic of re-submitting the bundles but other logic needn't be in a timeout. But on 2nd thought, thats overkill for now. I ll just use a timeout in the executor and await the future in the composer.

Reply:
i have done this! moved the timeout to executor and I am just awaiting the task in composer